rskj-core/src/test/java/co/rsk/peg/federation/FederationChangeIT.java

```diff
@@ -414,8 +527,138 @@ private void assertNewAndOldFederationsHaveExpectedAddress(
+            .getEntries().stream()
+            .map(Entry::getBtcTransaction)
+            .toList();
+
+        pegoutsTxs.forEach(
+            pegoutTx -> assertPegoutTxSigHashesAreSaved(pegoutTx, activations));
+    }
+
+    private void verifyPegoutTransactionCreatedEventWasEmitted(ActivationConfig.ForBlock activations) throws Exception {
+        var pegoutsTxs = bridgeStorageProvider.getPegoutsWaitingForConfirmations()
+            .getEntries().stream()
+            .map(Entry::getBtcTransaction)
+            .toList();
+
+        if (activations.isActive(ConsensusRule.RSKIP428)) {
+            pegoutsTxs.forEach(this::verifyPegoutTransactionCreatedEvent);
+        } else {
+            verify(bridgeEventLogger, never()).logPegoutTransactionCreated(any(), any());
+        }
+    }
+    
+    private void verifyPegouts(ActivationConfig.ForBlock activations) throws Exception {
+        var activeFederation = federationStorageProvider.getNewFederation(
+            BRIDGE_CONSTANTS.getFederationConstants(), activations);
+        var retiringFederation = federationStorageProvider.getOldFederation(
+            BRIDGE_CONSTANTS.getFederationConstants(), activations);
+
+        for (PegoutsWaitingForConfirmations.Entry pegoutEntry : bridgeStorageProvider.getPegoutsWaitingForConfirmations().getEntries()) {
+            var pegoutBtcTransaction = pegoutEntry.getBtcTransaction();
+            for (TransactionInput input : pegoutBtcTransaction.getInputs()) {
+                // Each input should contain the right scriptSig
+                var inputScriptChunks = input.getScriptSig().getChunks();
+                var inputRedeemScript = new Script(inputScriptChunks.get(inputScriptChunks.size() - 1).data);
+
+                // Get the standard redeem script to compare against, since it could be a flyover redeem script
+                var redeemScriptChunks = ScriptParser.parseScriptProgram(
+                    inputRedeemScript.getProgram());
+
+                var redeemScriptParser = RedeemScriptParserFactory.get(redeemScriptChunks);
+                var inputStandardRedeemScriptChunks = redeemScriptParser.extractStandardRedeemScriptChunks();
+                var inputStandardRedeemScript = new ScriptBuilder().addChunks(inputStandardRedeemScriptChunks).build();
+
+                Optional<Federation> spendingFederationOptional = Optional.empty();
+                if (inputStandardRedeemScript.equals(getFederationDefaultRedeemScript(activeFederation))) {
+                    spendingFederationOptional = Optional.of(activeFederation);
+                } else if (retiringFederation != null &&
+                    inputStandardRedeemScript.equals(getFederationDefaultRedeemScript(retiringFederation))) {
+                    spendingFederationOptional = Optional.of(retiringFederation);
+                } else {
+                    fail("Pegout scriptsig does not match any Federation");
+                }
```

Comment:
```suggestion
                Optional<Federation> spendingFederation = getSpendingFederation();
                assertTrue(spendingFederation.isPresent())
                
                private Optional<Federation> getSpendingFederation() {
                      if (inputStandardRedeemScript.equals(getFederationDefaultRedeemScript(activeFederation))) {
                               return activeFederation;
                      }
                      
                       if (retiringFederation != null &&
                    inputStandardRedeemScript.equals(getFederationDefaultRedeemScript(retiringFederation))) {
                               return retiringFederation;
                    }
                }
```

something like that, wdyt?

Reply:
btw, are we gonna have tests with pegouts from the retiring fed?